lib/macgyver.js

```diff
@@ -1,11 +1,88 @@
 var pristineEnv = require('./pristine-env');
 
+var pristineObject = pristineEnv().Object;
+var pristineArray = pristineEnv().Array;
+
+
+/*
+ * Copy an existing property from `from` to `to`.
+ * We don't want to pollute native prototypes with undefined properties,
+ * therefore we must check they exist first.
+ */
+function copy(to, from, key) {
+    if (from.hasOwnProperty(key)) {
+        to[key] = from[key];
+    }
+}
+
+/**
+ * Try to return a thing which prototype is in a pristine condition.
```

Comment:
Might be a few too many things in this sentence. Maybe something like:

Attempts to add native support for various methods to passed in object, without modifying the prototype.
If native support cannot be patched then the unmodified object is returned.
Currently only supports arrays.

Might be a bit formal though.